### PR TITLE
feat(get-starknet): add wallet_supportedWalletApi rpc method 

### PR DESCRIPTION
**Pull request by khanti42**

### CODE REVIEW

File: packages/get-starknet/src/rpcs/supported-wallet-api.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { SupportedWalletApi } from '../constants';
+import { StarknetWalletRpc } from '../utils/rpc';
+
+export type WalletSupportedWalletApiMethod = 'wallet_supportedWalletApi';
+type Params = RpcTypeToMessageMap[WalletSupportedWalletApiMethod]['params'];
+type Result = RpcTypeToMessageMap[WalletSupportedWalletApiMethod]['result'];
+
+export class WalletSupportedWalletApi extends StarknetWalletRpc {
+  async handleRequest(_param: Params): Promise<Result> {
+    return SupportedWalletApi;
```
Comment (stanleyyconsensys) [inline packages/get-starknet/src/rpcs/supported-wallet-api.ts:12]: i think the response format is not correct?

it is
``` 
   [{
      api_version: "some version"
   }]
```

File: packages/get-starknet/src/constants.ts [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
 
 // The supported RPC version should be update base on the Provider.
 // With Provider `Alchemy`, it is 0.7
 export const SupportedStarknetSpecVersion = ['0.7'];
+
+// The wallet API support is 0.7.2 but the RPC specs requests xx.yy. Hence we skip the last digits.
```
Comment (stanleyyconsensys) [inline packages/get-starknet/src/constants.ts:15]: i believe WalletApi is different with RPC
it is talking about the wrapper API version (wallet api versions)

File: packages/get-starknet/src/type.ts [hunk 3]
```diff
@@ -30,3 +30,8 @@ export type RequestSnapResponse = {
     blocked: boolean;
   };
 };
+
+export type ApiVersion = {
```
Comment (stanleyyconsensys) [inline packages/get-starknet/src/type.ts:34]: it seem useless?

File: packages/get-starknet/src/constants.ts [hunk 4]
```diff
@@ -1,10 +1,16 @@
+import type { ApiVersion } from './type';
```
Comment (stanleyyconsensys) [inline packages/get-starknet/src/constants.ts:1]: starknet.js contains that type

**Review by stanleyyconsensys — CHANGES_REQUESTED** [review]
Test is missing

File: packages/get-starknet/src/rpcs/supported-wallet-api.ts (hunk 1 above)
Comment (khanti42) [reply packages/get-starknet/src/rpcs/supported-wallet-api.ts:12]: Based on what I read here : https://github.com/starkware-libs/starknet-specs/blob/6d88b7399f56260ece3821c71f9ce53ec55f830b/wallet-api/wallet_rpc.json#L19 It seems to be an array of strings not array of object.

Comment (stanleyyconsensys) [reply packages/get-starknet/src/rpcs/supported-wallet-api.ts:12]: we can use string[] for now , but await for clarify

File: packages/get-starknet/src/constants.ts (hunk 2 above)
Comment (stanleyyconsensys) [reply packages/get-starknet/src/constants.ts:15]: we can use [0.7] for now , but await for clarify

File: packages/get-starknet/src/rpcs/supported-wallet-api.test.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+describe('WalletSupportedWalletApi', () => {
+  it('returns the supported wallet api version', async () => {
+    const wallet = createWallet();
+    mockWalletInit({ currentNetwork: SepoliaNetwork });
+
+    const walletSupportedWalletApi = new WalletSupportedWalletApi(wallet);
+    const result = await walletSupportedWalletApi.execute();
+
+    expect(result).toBe(SupportedWalletApi);
```
Comment (stanleyyconsensys) [inline packages/get-starknet/src/rpcs/supported-wallet-api.test.ts:13]:
```suggestion
    expect(result).toStrictEqual(SupportedWalletApi);
```

**Review by stanleyyconsensys — COMMENTED** [review]
a minor comment in the test

**Review by stanleyyconsensys — APPROVED** [review]

**Pull request merged**